cupy/indexing/insert.py

```diff
@@ -66,7 +67,77 @@ def put(a, ind, v, mode='wrap'):
     a.put(ind, v, mode=mode)
 
 
-# TODO(okuta): Implement putmask
+_putmask_kernel = core._kernel.ElementwiseKernel(
+    'Q mask, raw S values, uint64 len_vals', 'T out',
+    '''
+    if (mask) out = (T) values[i % len_vals];
+    ''',
+    'putmask_kernel'
+)
+
+
+def putmask(a, mask, values):
+    """
+    Changes elements of an array inplace, based on conditional mask and
+    input values.
+
+    Sets ``a.flat[n] = values[n]`` for each n where ``mask.flat[n]==True``.
+    If `values` is not the same size as `a` and `mask` then it will repeat.
+
+    Args
```

Comment:
This colon is necessary: https://google.github.io/styleguide/pyguide.html#doc-function-args
```suggestion
    Args:
```